### PR TITLE
chore(api): add descriptions and notes to components in Forms package

### PR DESCRIPTION
**Pull request by smritimadan**
This PR aims to restore and improve component documentation while following our content standards. Consideration was given to:

- providing information in a consistent, simple, clear and accessible way
- avoiding overloading with text or repetition
- using ways to make the content easier to scan quickly

## Checklist

- [x] :ok_hand: website updates are Garden Designer approved (add the designer as a reviewer)
- [x] :black_nib: copy updates are approved (add the content strategist as a reviewer)
- [x] :link: considered opportunities for adding cross-reference URLs (grep for keywords)
- [x] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

**Comment by jzempel** [comment]
Hey @smritimadan, I noticed some canonical structures in the `Radio` component class (incl. Radio, Checkbox, Toggle) that are not ideal. The site has corresponding example code that needs to be finessed. Let's plan to catch-up on the details here.

tldr; the only component that a `<Radio>` (or others) should wrap is `<Label>`.

**Comment by smritimadan** [comment]
> canonical structures in the `Radio` component class (incl. Radio, Checkbox, Toggle) that are not ideal. The site has corresponding example code that needs to be finessed. 

>tldr; the only component that a <Radio> (or others) should wrap is <Label>.

Gotcha. To be clear, should the `Hint` be a sibling to `<Radio>` instead of a child?

**Comment by jzempel** [comment]
> Gotcha. To be clear, should the `Hint` be a sibling to `<Radio>` instead of a child?

yes, along with `Message` as a sibling. Same with `Checkbox` and `Toggle`.
